### PR TITLE
Fix: Lint rule issues strict-boolean-expressions

### PR DESCRIPTION
**Pull request by lmuntanerDfinity**
# Motivation

Fix lint issues regarding the rule `@typescript-eslint/strict-boolean-expressions`

# Changes

* Fix issues with `@typescript-eslint/strict-boolean-expressions` in many files.

# Tests

No new tests. Thankful for the current tests.

### CODE REVIEW

**Comment by lmuntanerDfinity** [comment]
@mstrasinskis @peterpeterparker please review

**Comment by lmuntanerDfinity** [comment]
Current errors:  "60 problems (26 errors, 34 warnings)"

File: frontend/svelte/src/lib/components/proposals/ProposalCard.svelte [hunk 1]
```diff
@@ -47,13 +47,15 @@
...
       <Badge slot="end" {color}
-        >{status ? $i18n.status[ProposalStatus[status]] : ""}</Badge
+        >{status !== undefined
+          ? $i18n.status[ProposalStatus[status]]
+          : ""}</Badge
       >
 
       <p class="info"><Proposer {proposalInfo} /></p>
-      <p class="info"><small>Id: {id || ""}</small></p>
+      <p class="info"><small>Id: {id === undefined ? "" : id}</small></p>
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/proposals/ProposalCard.svelte:58]: would `{id ?? ""}` fix it?

File: frontend/svelte/src/lib/components/common/Banner.svelte [hunk 2]
```diff
@@ -6,13 +6,14 @@
 
   const localstorageKey: string = "nnsdapp-testnet-banner-display";
 
+  const visibleStorage: string | null = localStorage.getItem(localstorageKey);
   let visible: boolean = JSON.parse(
-    localStorage.getItem(localstorageKey) || "true"
+    visibleStorage === null ? "true" : visibleStorage
```
Comment (peterpeterparker) [inline frontend/svelte/src/lib/components/common/Banner.svelte:11]: would `visibleStorage ?? true` fix it?

**Review by peterpeterparker — COMMENTED** [review]
can nullish coalescing solve some?

File: frontend/svelte/src/lib/components/proposals/ProposalCard.svelte (hunk 1 above)
Comment (lmuntanerDfinity) [reply frontend/svelte/src/lib/components/proposals/ProposalCard.svelte:58]: yes! Thanks! I thought it wouldn't. Redoing all the changes!

**Comment by lmuntanerDfinity** [comment]
@peterpeterparker thanks for the review, I refactored and it's much cleaner now!

**Review by peterpeterparker — APPROVED** [review]
Cool quite happy it worked out. Thanks for the PR!

**Pull request merged**